Fix/android compatibility

DottL opened the pull request:
# Fix Android Compatibility Issues
## Overview
Resolves multiple Android-specific issues including build failures, Worklets initialization errors, and coordinate scaling problems for parking lot markers. Configures the monorepo build system to work with pnpm's virtual store structure and modernizes gesture handling to use the current Reanimated API.

## Changes Made
### Added React Native Reanimated Dependencies

- Added `react-native-reanimated@^4.2.1` for modern gesture handling
- Added `react-native-worklets@^0.7.1` as peer dependency
- Configured Babel plugin for Reanimated

### Migrated to Modern Gesture API

- Replaced deprecated PanGestureHandler and PinchGestureHandler
- Implemented new Reanimated Gesture API
- Maintains same pan and pinch-to-zoom functionality

### Fixed Coordinate Scaling for Cross-Platform Compatibility

- Created mapUtils.ts with coordinate conversion utilities
- Convert hardcoded iPhone positions to proportional coordinates
- Ensures parking lot markers display correctly across different screen sizes

### Fixed pnpm Monorepo Compatibility

- Added `.npmrc` with `public-hoist-pattern` to hoist React Native packages
- Updated `settings.gradle` to reference hoisted `@react-native/gradle-plugin`
- Updated `app/build.gradle` paths for `reactNativeDir`, `codegenDir`, and `cliFile`
- Fixed `react-native-vector-icons` path in gradle config

**Why These Changes Were Needed:**
- `pnpm` uses a virtual store structure (`.pnpm/package@version_hash/`) instead of flat `node_modules/`; 
  - Android's Gradle build system expects native dependencies at predictable paths; Gradle couldn't find React Native packages at the expected paths, causing build failures. 
- The solution hoists React Native packages to predictable locations.
  - The `public-hoist-pattern` configuration instructs pnpm to create symlinks for React Native packages at standard `node_modules/` locations, making them accessible to Gradle; This allows gradle to reference stable maintainable paths across dependency updates.

## Files Changed
### Added Files:
- `.npmrc` - pnpm hoisting configuration
- `apps/mobile/src/utils/mapUtils.ts` - coordinate conversion utilities

### Modified Files:
- `pnpm-lock.yaml` - updated dependencies
- `apps/mobile/package.json` - added reanimated & worklets packages
- `apps/mobile/babel.config.js` - added reanimated plugin
- `apps/mobile/android/settings.gradle` - updated gradle plugin paths
- `apps/mobile/android/app/build.gradle` - updated React Native & vector-icons paths
- `apps/mobile/src/screens/MapScreen.tsx` - migrated to modern Gesture API
- `apps/mobile/src/data/mockParkingLots.ts` - updated marker coordinates to use proportional system
- `apps/mobile/jest.setup.js` - added manual mocks for Reanimated and enhanced gesture handler mocks

## Breaking Changes
- None - maintains backward compatibility with existing functionality

DottL committed:
feat: add react-native-reanimated dependency

DottL committed:
feat: configure react-native-vector-icons for Android

DottL committed:
feat: add reanimated babel plugin

DottL committed:
feat: migrate to Reanimated gestures and fix Android scaling

DottL committed:
chore: add pnpm config to hoist React Native packages

DottL committed:
fix(android): use hoisted packages in gradle paths

DottL committed:
fix: add manual mock for react-native-reanimated in Jest setup

chuckmilton commented on apps/mobile/src/utils/mapUtils.ts:
```diff
@@ -0,0 +1,15 @@
+
+const { width: SCREEN_WIDTH } = Dimensions.get('window');
+
+const MAP_IMAGE_SIZE = 1098;
```
We should document this. Hard to understand without any context

chuckmilton commented on apps/mobile/src/screens/MapScreen.tsx:
Minor issue: Incomplete Jest Mock

The mock is missing `minPointers`, `maxPointers`, and `Gesture.Simultaneous`:
```typescript
Gesture.Pan().minPointers(1).maxPointers(1) // Not mocked
Gesture.Simultaneous(...) // Not mocked
```

This won't break tests currently, but if someone writes a test that directly invokes these methods, it will fail.

chuckmilton commented on apps/mobile/src/screens/MapScreen.tsx:
```diff
@@ -73,6 +74,47 @@ const MapScreen: React.FC = () => {
+    .minPointers(1)
+    .maxPointers(1)
+    .onUpdate((e) => {
+      translateX.value = savedTranslateX.value + e.translationX;
```
This pan gesture has no limits. If users are able to pan the map infinitely off-screen, then we should try to prevent that

chuckmilton commented on apps/mobile/src/screens/MapScreen.tsx:
```typescript
// mapUtils.ts
const MAP_DISPLAY_SIZE = SCREEN_WIDTH * 2.5;  // Scaling uses 2.5x

// MapScreen.tsx styles
mapImage: {
  width: screenWidth * 2.25,   // But image displays at 2.25x
  height: screenWidth * 2.25,
}
```

These multipliers should match. Currently, lot positions are calculated for a 2.5x canvas but rendered on a 2.25x image, which could cause markers to appear offset. We should unify to a single constant.

chuckmilton reviewed (approved):
Great work fixing Android compatibility

Left one comment about the coordinate scaling multiplier mismatch (2.5x vs 2.25x) that should be unified to avoid marker offset. Minor suggestions on Jest mocks and documentation. Otherwise LGTM

lawrenceDegoma reviewed (approved):
Charles got all the points of suggestion but otherwise all good work!

DottL committed:
refactor: extract map constants to theme & update lot positions

DottL committed:
fix: complete gesture handler mock w/ minPointers, maxPointers, & Simultaneous

DottL committed:
fix: add focal-point pinch zoom & pan bounds to map

- Implement focal-point-aware pinch zoom that scales around touch position
- Updated map image to increase view boundary
- Update parking lot coordinates to align with updated map image
- Implement pan while pinching
- Remove unused screenHeight dimension + unused min/maxPointers mock methods
